### ISSUE TITLE
T-SQL: INSERT INTO <> DEFAULT VALUES cannot be parsed

### ISSUE THREAD

**Issue by borchero** [sqlfluff/sqlfluff#3933]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

The query below could not be parsed.

### Expected Behaviour

The query should have been parsed correctly.

### Observed Behaviour

`sqlfluff lint -d tsql` results in the following issue:

```
== [test.sql] FAIL
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'INSERT
                       | INTO mytable WITH(TABLOCKX)\nDEFAU...'
WARNING: Parsing errors found and dialect is set to 'tsql'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### How to reproduce

```sql
INSERT INTO mytable
DEFAULT VALUES;
```

### Dialect

`tsql`

### Version

`1.3.2`

### Configuration

n/a

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
